### PR TITLE
DR-1788 - computeSnapshot - specify dataset vs. snapshot firestore

### PR DESCRIPTION
**Pull request by snf2ye**

### CODE REVIEW

File: datarepo-clienttests/src/main/resources/servers/shdev.json [hunk 1]
```diff
@@ -16,6 +16,6 @@
   },
   "testRunnerServiceAccountFile": "jade-k8-sa.json",
   "skipDeployment": true,
-  "skipKubernetes": false,
```
Comment (snf2ye) [inline datarepo-clienttests/src/main/resources/servers/shdev.json:19]: This is just for my personal dev deploy setup - I've been meaning to change this, so I'm going to leave this in.

File: src/main/java/bio/terra/service/filedata/google/firestore/FireStoreDao.java [hunk 2]
```diff
@@ -437,7 +442,7 @@ FireStoreDirectoryEntry computeDirectory(
             for (Map.Entry<String, List<FireStoreDirectoryEntry>> entry : fileRefsByDatasetId.entrySet()) {
                 // Retrieve the file metadata from Firestore
                 final List<FireStoreFile> fireStoreFiles =
-                    fileDao.batchRetrieveFileMetadata(firestore, entry.getKey(), entry.getValue());
+                    fileDao.batchRetrieveFileMetadata(datasetFirestore, entry.getKey(), entry.getValue());
```
Comment (snf2ye) [inline src/main/java/bio/terra/service/filedata/google/firestore/FireStoreDao.java:445]: This is the main change! It was getting passed the snapshot Firestore before.

File: src/main/java/bio/terra/service/filedata/google/firestore/FireStoreDirectoryDao.java [hunk 3]
```diff
@@ -200,7 +199,7 @@ private FireStoreDirectoryEntry runTransactionWithRetry(Firestore firestore,
                                                             LookupFunction lookupFunction,
                                                             String transactionOp,
                                                             String warnMessage) throws InterruptedException {
-        for (int i = 0; i < RETRIES; i++) {
+        for (int i = 0; i < LOOKUP_RETRIES; i++) {
```
Comment (snf2ye) [inline src/main/java/bio/terra/service/filedata/google/firestore/FireStoreDirectoryDao.java:202]: This should fix a second bug we were running into - We were getting some timeouts on file ingest, but hopefully bumping the number of retries up should help.

File: datarepo-clienttests/src/main/resources/servers/shdev.json [hunk 4]
```diff
@@ -16,6 +16,6 @@
...
+  "skipKubernetes": true,
```
Comment (nmalfroy) [inline datarepo-clienttests/src/main/resources/servers/shdev.json:19]: 👍 I did the same

File: src/main/java/bio/terra/service/filedata/google/firestore/FireStoreDao.java [hunk 5]
```diff
@@ -162,15 +165,15 @@ public void snapshotCompute(Snapshot snapshot) throws InterruptedException {
             List<FireStoreDirectoryEntry> updateBatch = new ArrayList<>();
 
             String retrieveTimer = performanceLogger.timerStart();
-            computeDirectory(firestore, snapshotId, topDir, updateBatch);
+            computeDirectory(snapshotFirestore, datasetFirestore, snapshotId, topDir, updateBatch);
```
Comment (nmalfroy) [inline src/main/java/bio/terra/service/filedata/google/firestore/FireStoreDao.java:168]: Getting a head start on Q3 metadata work I see :)

File: src/main/java/bio/terra/service/filedata/google/firestore/FireStoreDao.java (hunk 2 above)
Comment (nmalfroy) [reply src/main/java/bio/terra/service/filedata/google/firestore/FireStoreDao.java:445]: gah, figures

**Review by nmalfroy — APPROVED** [review]
AWESOME job fishing this out.  Were you able to test manually?  We should remember in a followon to add some automated testing around this

**Comment by snf2ye** [comment]
@nmalfroy 
> AWESOME job fishing this out. Were you able to test manually? We should remember in a followon to add some automated testing around this

- Thanks! 
- Yes, we did some manual testing. The alpha test passed locally (locally => these changes deployed to my personal dev env that is set up to act like prod). We also looked did some manual digging around firestore to make sure everything looked okay after the change.
- I was thinking we couldn't really test this until we had RBS setup.

**Review by samanehsan — APPROVED** [review]
LGTM! :octocat:

**Pull request merged**